### DISCUSSION TITLE
How to save the ExplainerDataset?

### DISCUSSION THREAD

**Discussion by CXX1113** [pyg-team/pytorch_geometric#8294] [Q&A]
Such as 
` 
dataset = ExplainerDataset(
graph_generator=BalancedBinaryTree(num_levels=8),
motif_generator=CycleMotif(num_nodes=6),
num_motifs=80,
transform=MultiConstantFeat(value=1., num_dim=10, norm=True),)
`
The purpose of saving the data set is to keep the generated data set unchanged the next time you run the program.

**Comment by akihironitta** [comment]
AFAIK, there's no interface that PyG provides specifically to save generated data, so you have to implement it on your side. I think you can utilise `self.save(...)` to save data like other `InMemoryDataset`s, e.g., https://github.com/pyg-team/pytorch_geometric/blob/fd7ff50287e6bd251247804f15e0ae96f866ddac/torch_geometric/datasets/movie_lens_1m.py

**Comment by rusty1s** [comment]
You can also just set a seed so that data is equvialent across runs. Otherwise, you can just do
```python
torch.save(dataset, 'dataset.pt')

dataset = torch.load('dataset.pt')
